config/passport.js

```diff
@@ -1,6 +1,22 @@
 const GoogleStrategy = require('passport-google-oauth').OAuth2Strategy;
 const LinkedInStrategy = require('passport-linkedin-oauth2').Strategy;
 
+const StrategyCallback = (accessToken, refreshToken, profile, done) => {
+  const {
+    id,
+    name: { givenName, familyName },
+    emails,
+  } = profile;
+  const userData = {
+    id: id,
+    email: emails[0].value,
+    firstName: givenName,
+    lastName: familyName,
+    token: accessToken,
+  };
```

Comment:
rewrite db structure to apply external users and make password field non-nullable